applications/serial_lte_modem/src/slm_at_udp_proxy.c

```diff
@@ -191,6 +192,25 @@ static int do_udp_client_connect(const char *url, uint16_t port)
 			ret = -errno;
 			goto cli_exit;
 		}
+
+		/* Best effort to enable/disable DTLS conn_id and load it */
+		uint32_t dtls_cid = (proxy.conn_id != 0) ?
+			NRF_SO_SEC_DTLS_CID_ENABLED : NRF_SO_SEC_DTLS_CID_DISABLED;
+
+		ret = nrf_setsockopt(proxy.sock, NRF_SOL_SECURE, NRF_SO_SEC_DTLS_CID,
+				     &dtls_cid, sizeof(dtls_cid));
+		if (ret) {
+			LOG_WRN("set conn id (%d) failed: %d", dtls_cid, -errno);
+		} else if (proxy.conn_id != 0) {
+			ret = nrf_setsockopt(proxy.sock, NRF_SOL_SECURE,
+					     NRF_SO_SEC_DTLS_CONN_LOAD,
+					     NULL, 0);
+			if (ret) {
+				LOG_WRN("Load DTLS connection failed: %d", ret);
+			} else {
+				LOG_INF("DTLS connection resumed");
+			}
+		}
```

Comment:
My proposal would be not to use to storing/loading, at least not in the first phase.

Either drop it, or have it as a separate option / AT command.

Reply:
I don't think modem will automatically SAVE/LOAD CID so if APP does not do that, CID might not take effect for DTLS re-handshake.

Reply:
Same CID should not be re-used in DTLS re-handshake.

https://datatracker.ietf.org/doc/html/rfc9146:

In DTLS 1.2, CIDs are exchanged at the beginning of the DTLS session only. There is no dedicated "CID update" message that allows new CIDs to be established mid-session, because DTLS 1.2 in general does not allow TLS 1.3-style post-handshake messages that do not themselves begin other handshakes. **When a DTLS session is resumed or renegotiated, the "connection_id" extension is negotiated afresh.**

Also:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/nrf_modem/doc/sockets.html

NRF_SO_SEC_DTLS_CONN_SAVE
Save DTLS connection. This option is write-only. This option require a DTLS v1.2 connection with renegotiation disabled.